### PR TITLE
Code quality: Refactor `useQuickActions`

### PR DESCRIPTION
**Pull request by miina**
## Context
Splits the `useQuickActions` file into separate hooks for making it easier to manage.

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partial for #12705

### CODE REVIEW

File: packages/story-editor/src/app/quickActions/useQuickActions.js [hunk 1]
```diff
@@ -938,27 +260,21 @@ const useQuickActions = () => {
...
+    case ElementType.Text:
       return textActions;
-    case ELEMENT_TYPES.VIDEO:
+    case ElementType.Video:
       return videoActions;
-    case ELEMENT_TYPES.STICKER:
-      return foregroundCommonActions;
     default:
       return [];
   }
```
Comment (barklund) [inline packages/story-editor/src/app/quickActions/useQuickActions.js:280]: 280 lines versus 964... I 🎺  it!

File: packages/story-editor/src/app/quickActions/useForegroundActions.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import { useStory } from '../story';
+import { ACTIONS } from './constants';
+import useElementReset from './useElementReset';
+
+function useForegroundActions({
+  actionProps,
+  selectedElement,
+  handleFocusPanel,
+  resetProperties,
+}) {
```
Comment (barklund) [inline packages/story-editor/src/app/quickActions/useForegroundActions.js:37]: I tried typing this one for fun, and it was  basically impossible. It handles so many different types of elements but implicitly through the reset properties value, that it's really hard to do. I think it needs to be split into resets for the different element types, which in turn of course can use the same underlying utils, but not directly like this.

**Review by barklund — APPROVED** [review]
🪘

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
There's 1 failing test:

```
useQuickActions › background video element is selected › clicking `clear filters and animations` should update the element

    TypeError: handleElementReset is not a function
```

**Pull request merged**